docs/compiler.md

```diff
@@ -181,6 +181,64 @@ Deployment works via an RPC server, an address of which is passed via `-r`
 option and should be signed using a wallet from `-w` option. More details can
 be found in `deploy` command help.
 
+#### Config file
+Configuration file contains following options:
+
+| Parameter | Description | Example |
+| --- | --- | --- |
+| `name` | Contract name in the manifest. | `"My awesome contract"`
+| `safemethods` | List of methods which don't change contract state | `["balanceOf", "decimals"]`
+| `supportedstandards` | List of standards this contract implement. Current options: `NEP-11`, `NEP-17`, `NEP-11-Paylable`, `NEP-17-Payable`. This will enable additional checks in compiler. | `["NEP-17"]`
+| `events` | Events emitted by this contract. | See [Events](#Events). |
+| `permissions` | Foreign calls allowed for this contract. | See [Permissions](#Permissions). |
+
+##### Events
+Each event must have a name and 0 or more parameters. Parameters are specified using their name and type.
+
+As an example consider `Transfer` event from `NEP-17` standard:
+```
+- name: Transfer
+  parameters:
+    - name: from
+      type: Hash160
+    - name: to
+      type: Hash160
+    - name: amount
+      type: Integer
+```
+
+##### Permissions
+Permissions restrict set of foreign methods this contract is allowed to call.
+By default, no calls are allowed. Simplest permission is to allow everything:
+```
+- methods: '*'
+```
+
+Another common case is to allow calling `onNEP17Payment`, which is necessary if your
+contract can transfer NEP17 tokens to other contracts:
```

Comment:
It's more for NEP-17 token implementations (that call `onNEP17Payment` of other contracts) rather than contracts that own/manage assets (these call `transfer`).